rancher2/structure_cluster_rke_config_services_kube_api.go

```diff
@@ -210,6 +263,55 @@ func expandClusterRKEConfigServicesKubeAPIAuditLog(p []interface{}) (*management
 	return obj, nil
 }
 
+func expandClusterRKEConfigServicesKubeAPIAdmissionConfigurationPlugins(p []interface{}) ([]interface{}, error) {
+	obj := []interface{}{}
+	if len(p) == 0 || p[0] == nil {
+		return obj, nil
+	}
+
+	for i, plugin := range p {
+		if v, ok := plugin.(map[string]interface{}); ok && len(v) > 0 {
+			obj = append(obj, make(map[string]interface{}))
+			if o, ok := (obj[i]).(map[string]interface{}); ok {
```

Comment:
@HarrisonWAffel I don't know if this is the most elegant way to do this cast.

Reply:
In Go we want to avoid nesting if we can, I think the following might be cleaner 

```go
for i, plugin := range p { 
	pluginMap, ok := plugin.(map[string]interface{})
	if !ok || len(pluginMap) == 0 {
		continue // or return, if this is an error condition
	}
	var newObj map[string]interface{}
	newObj["name"] = pluginMap["name"].(string)
	newObj["path"] = pluginMap["path"].(string)
	configuration, err := ghodssyamlToMapInterface(pluginMap["configuration"].(string))
	if err != nil {
		return obj, fmt.Errorf("Unmarshalling plugin configuration from yaml for %v: %v", pluginMap["name"], err)
	}
	newObj["configuration"] = configuration
	obj = append(obj, newObj)
}
```

Reply:
Done Thanks for the tips!